### DISCUSSION TITLE
Define a generic schema for tasks

### DISCUSSION THREAD

**Discussion by hooverdc** [RealOrangeOne/django-tasks#71] [Ideas]
Background: I'm implementing my own SQS backend and worker for tasks. It would be nice to have a generic schema defined to serialize and deserialize tasks to and from. I think this would cut down on idiosyncrasies between each backend implementation. This could also lead itself to a generic worker implementation that reads from whatever backend.

For example
```
{
  func: my_app.func,
  args: [],
  kwargs: {},
  attrs: {
    priority: 1,
    run_after: 2024-12-31,
  },
}
```

**Comment by RealOrangeOne** [comment]
The generic schema for tasks is the `Task` and `TaskResult` objects. Exactly what a backend supports for serialization is entirely backend-specific. Sure, we could force everything down to JSON, but for some backends that isn't necessary.

A backend can just accept the `Task` and `TaskResult` classes, and do with them whatever they need to. Their types are well-defined, and changes won't be backwards-incompatible.
